core/controlplane/config/templates/cloud-config-controller

```diff
@@ -791,6 +791,119 @@ write_files:
       sed -i -e "s#\$ETCDKEY#$etcd_key#g" /srv/kubernetes/manifests/calico.yaml
 
 {{ end }}
+{{ if .ExportKubeResources }}
+  - path: /srv/kubernetes/manifests/cluster-dump.yaml
+    content: |
+      ---
+      apiVersion: extensions/v1beta1
+      kind: Deployment
+      metadata:
+        name: cluster-dump
+        namespace: kube-system
+        labels:
+          k8s-app: cluster-dump-policy
+      spec:
+        replicas: 1
+        template:
+          metadata:
+            name: cluster-dump
+            namespace: kube-system
+            labels:
+              k8s-app: cluster-dump-policy
+          spec:
+            containers:
+            - name: cluster-dump-dumper
+              image: {{.HyperkubeImage.RepoWithTag}}
+              command: ["/bin/bash", "-c" ]
+              args:
+                - |
+                    set -x ;
+                    DUMP_DIR_COMPLETE=/cluster-dump/complete ;
+                    mkdir -p ${DUMP_DIR_COMPLETE} ;
+                    while true; do
+                      TIMESTAMP=$(date +%Y-%m-%d_%H-%M-%S)
+                      DUMP_DIR=/cluster-dump/tmp/${TIMESTAMP} ;
+                      mkdir -p ${DUMP_DIR} ;
+                      RESOURCES_OUT_NAMESPACE=( namespaces persistentvolumes nodes storageclasses ) ;
+                      for r in ${RESOURCES_OUT_NAMESPACE[@]};do
+                        echo " Searching for resources: ${r}" ;
+                        /kubectl get --export -o=json ${r} | \
+                        jq '.items[] |
+                            del(.status,
+                            .metadata.uid,
+                            .metadata.selfLink,
+                            .metadata.resourceVersion,
+                            .metadata.creationTimestamp,
+                            .metadata.generation,
+                            .spec.claimRef
+                          )' > ${DUMP_DIR}/${r}.json ;
+                      done ;
+                      RESOURCES_IN_NAMESPACE=( componentstatuses configmaps daemonsets deployments endpoints events horizontalpodautoscalers
+                      ingresses jobs limitranges networkpolicies  persistentvolumeclaims pods podsecuritypolicies podtemplates replicasets
+                      replicationcontrollers resourcequotas secrets serviceaccounts services statefulsets thirdpartyresources ) ;
```

Comment:
Did you miss `clusterrolebindings`, `clusterroles`, `namespaces`, `poddisruptionbudgets`, `rolebindings`, `roles`, `storageclasses`?
How does it work for `persistentvolumes`? Are `persistentvolumes` able to be backed up and restored?